src/lib/azure/updatetable.ts

```diff
@@ -0,0 +1,279 @@
+      if (error) {
+        logger.error(error.message);
+      } else {
+        logger.info(`Added new entry for ${filterName} ${filterValue}`);
+      }
+    }
+  );
+  return newEntry;
+};
+
+/**
+ * Updates an existing deployment given the existing entries in the table for the filters
+ * @param entries Existing entries in the deployment table
+ * @param accountName Name of the storage account
+ * @param accountKey Access key for the storage account
+ * @param tableName Name of the table in storage account to use
+ * @param partitionKey Partition key for the project
+ * @param filterName Name of the field to filter by, for eg. imageTag is a filter name for the image tag release pipeline
+ * @param filterValue Value of the filter key
+ * @param key1 Name of the first field being inserted to the table
+ * @param value1 Value of the first field being inserted to the table
+ * @param key2 Name of the second field being inserted to the table
+ * @param value2 Value of the second field being inserted to the table
+ * @param key3 Name of the third field being inserted to the table
+ * @param value3 Value of the third field being inserted to the table
+ */
+export const updateExistingDeployment = <T>(
+  entries: any[],
+  accountName: string,
+  accountKey: string,
+  tableName: string,
+  partitionKey: string,
+  filterName: string,
+  filterValue: string,
+  key1: string,
+  value1: string,
+  key2?: string,
+  value2?: string,
+  key3?: string,
+  value3?: string
+): any => {
+  let addEntity = false;
+  if (entries.length !== 0) {
+    const entry: any = entries[0];
+    if (key1 in entry && entry[key1]._ !== value1.toLowerCase()) {
+      addEntity = true;
+    }
+    entry[key1] = value1.toLowerCase();
+
+    if (key2 && value2) {
+      if (key2 in entry && entry[key2]._ !== value2.toLowerCase()) {
+        addEntity = true;
+      }
+      entry[key2] = value2.toLowerCase();
+    }
+
+    if (key3 && value3) {
+      if (key3 in entry && entry[key3]._ !== value3.toLowerCase()) {
+        addEntity = true;
+      }
+      entry[key3] = value3.toLowerCase();
+    }
+
+    const tableService = azure.createTableService(accountName, accountKey);
+    if (addEntity) {
+      entry.RowKey = getRowKey();
+      insertToTable(
+        accountName,
+        accountKey,
+        tableName,
+        entry,
+        (err, res, response) => {
+          if (err) {
+            logger.error(err.message);
+          } else {
+            logger.info(`Added new entry for ${filterName} ${filterValue}`);
+          }
+        }
+      );
+    } else {
+      updateEntryInTable(
+        accountName,
+        accountKey,
+        tableName,
+        entry,
+        (err, res, response) => {
+          if (err) {
+            logger.error(err.message);
+          } else {
+            logger.info(
+              `Updated existing entry for ${filterName} ${filterValue}`
+            );
+          }
+        }
+      );
+    }
```

Comment:
How about moving this to try/catch block to reduce it to one `err` check and logging.